### PR TITLE
Closes #1079: Managing site permissions exceptions

### LINKED ISSUE #1079: FNX-4781 ⁃ Managing exceptions of a site permissions

### PR DESCRIPTION
**Pull request by Amejia481**
[Video demo](https://drive.google.com/file/d/1SZkALa15r2ho1gyjtuHnitwcGqIlcnwW/view?usp=sharing) 🤠 .

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

**Comment by Amejia481** [comment]
It looks like TC has the latest snapshot but Bitrise doesn't

**Comment by boek** [comment]
rebuilt bitrise cache

**Comment by Amejia481** [comment]
Thanks @boek !

**Comment by colintheshots** [comment]
Sorry, I wrote my review like 12 hours ago and thought I submitted it. It was in draft still.

File: app/src/main/java/org/mozilla/fenix/settings/Extensions.kt [hunk 1]
```diff
@@ -73,3 +76,37 @@ fun SitePermissions.toggle(featurePhone: PhoneFeature): SitePermissions {
...
+fun PhoneFeature.getPreferenceKey(context: Context): String {
+    return when (this) {
+        PhoneFeature.CAMERA -> context.getString(R.string.pref_key_phone_feature_camera)
+        PhoneFeature.LOCATION -> context.getString(R.string.pref_key_phone_feature_location)
+        PhoneFeature.MICROPHONE -> context.getString(R.string.pref_key_phone_feature_microphone)
+        PhoneFeature.NOTIFICATION -> context.getString(R.string.pref_key_phone_feature_notification)
+    }
+}
+
+fun initBockedByAndroidView(phoneFeature: PhoneFeature, bockedByAndroidView: View) {
```
[the remark this answers is not in the document]
Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/settings/Extensions.kt]: Fixed

File: app/src/main/java/org/mozilla/fenix/settings/PhoneFeature.kt [hunk 2]
```diff
@@ -13,16 +13,16 @@ import mozilla.components.feature.sitepermissions.SitePermissions
...
-    CAMERA(SitePermissionsManagePhoneFeature.CAMERA_PERMISSION, arrayOf(Manifest.permission.CAMERA)),
-    LOCATION(
-        SitePermissionsManagePhoneFeature.LOCATION_PERMISSION, arrayOf(
-            ACCESS_COARSE_LOCATION,
-            ACCESS_FINE_LOCATION
-        )
-    ),
-    MICROPHONE(SitePermissionsManagePhoneFeature.MICROPHONE_PERMISSION, arrayOf(RECORD_AUDIO)),
-    NOTIFICATION(SitePermissionsManagePhoneFeature.NOTIFICATION_PERMISSION, emptyArray());
+    CAMERA(ID_CAMERA_PERMISSION, arrayOf(Manifest.permission.CAMERA)),
```
[the remark this answers is not in the document]
Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/settings/PhoneFeature.kt]: Fixed

File: app/src/main/java/org/mozilla/fenix/settings/SitePermissionsDetailsExceptionsFragment.kt [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        job = Job()
+    }
+
+    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
+        setPreferencesFromResource(R.xml.site_permissions_details_exceptions_preferences, rootKey)
+    }
+
+    override fun onResume() {
+        super.onResume()
+        launch {
```
[the remark this answers is not in the document]
Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/settings/SitePermissionsDetailsExceptionsFragment.kt]: Fixed

**Comment by Amejia481** [comment]
@colintheshots I updated the PR with the suggestions!

**Pull request merged**